distributed/client.py

```diff
@@ -1288,6 +1288,43 @@ def run(self, function, *args, **kwargs):
         """
         return sync(self.loop, self._run, function, *args, **kwargs)
 
+    @gen.coroutine
+    def _run_coroutine(self, function, *args, **kwargs):
+        workers = kwargs.pop('workers', None)
+        responses = yield self.scheduler.broadcast(msg=dict(op='run_coroutine',
+                                                function=dumps(function),
+                                                args=dumps(args),
+                                                kwargs=dumps(kwargs)),
+                                                workers=workers)
+        results = {}
+        for key, resp in responses.items():
+            if resp['status'] == 'OK':
+                results[key] = resp['result']
+            elif resp['status'] == 'error':
+                raise loads(resp['exception'])
+        raise Return(results)
+
+    def run_coroutine(self, function, *args, **kwargs):
+        """
+        Spawn a coroutine on all workers.
+
+        This spaws a coroutine on all currently known workers and then waits
+        for the coroutine on each worker.  The coroutines' results are returned
+        as a dictionary keyed by worker address.
+
+        Parameters
+        ----------
+        function: a coroutine function
+            (typically a function wrapped in gen.coroutine or
+             a Python 3.5+ async function)
+        *args: arguments for remote function
+        **kwargs: keyword arguments for remote function
+        workers: list
+            Workers on which to run the function. Defaults to all known workers.
+
+        """
+        return sync(self.loop, self._run_coroutine, function, *args, **kwargs)
```

Comment:
Is it possible to check the input function to see if it is a coroutine and, if so, use the `Client.run` method for both cases?